lucet-concurrency-tests/tests/killswitch.rs

```diff
@@ -0,0 +1,731 @@
+use lucet_runtime::vmctx::Vmctx;
+use lucet_runtime::{
+    lucet_hostcall, Error, Instance, InstanceHandle, KillError, KillSuccess, Limits, Region,
+    RunResult, TerminationDetails, TrapCode,
+};
+use std::sync::Arc;
+use std::thread;
+
+use lucet_module::FunctionPointer;
+use lucet_runtime::MmapRegion;
+use lucet_runtime_internals::lock_testpoints::{SyncWaiter, Syncpoint};
+use lucet_runtime_internals::module::Module;
+use lucet_runtime_internals::module::{MockExportBuilder, MockModuleBuilder};
+use lucet_runtime_internals::vmctx::lucet_vmctx;
+use lucet_runtime_tests::build::test_module_c;
+use lucet_runtime_tests::helpers::test_ex;
+use lucet_runtime_tests::helpers::test_nonex;
+
+static mut ENTERING_GUEST: Option<Syncpoint> = None;
+
+/// A convenience wrapper around running our mock timeout module's `onetwothree` function,
+/// and asserting that it returned the expected result.
+fn run_onetwothree(inst: &mut Instance) {
+    let retval = inst
+        .run("onetwothree", &[])
+        .expect("instance runs")
+        .unwrap_returned();
+    assert_eq!(libc::c_int::from(retval), 123);
+}
+
+pub fn mock_killswitch_module() -> Arc<dyn Module> {
```

Comment:
It feels like the purpose of this module is to provide a function that represents every step of the KillSwitch lifecycle.  Do I have that right?

Reply:
Yep! Additionally, functions to distinguish cases we want to test even though they are the same with respect to the KillSwitch lifecycle: this exposes a hostcall that does nothing, and a hostcall that yields so we can test both, even though they're both just hostcalls to `KillSwitch`.